### PR TITLE
WIP Add translator to get task ID.

### PR DESCRIPTION
**Pull request by Chadius**
- Made Git Tracker Name Translator, whose mission is to convert from git branch names to task IDs.

### CODE REVIEW

File: git.go [hunk 1]
```diff
@@ -9,4 +9,4 @@ import (
 func GetBranchName() string {
 	output, _ := exec.Command("git", "rev-parse", "--abbrev-ref", "HEAD").Output()
 	return strings.TrimSpace(string(output))
-}
+}
```
Comment (carpeliam) [inline git.go]: Any reason to remove this newline? There might be some good reasons to keep it - https://thoughtbot.com/blog/no-newline-at-end-of-file

File: git_test.go [hunk 2]
```diff
@@ -14,10 +14,10 @@ var _ = Describe("Git", func() {
 	path := os.Getenv("PATH")
 
 	BeforeEach(func() {
-		os.Setenv("PATH", fmt.Sprintf("./fixtures:%s", path))
+		_ = os.Setenv("PATH", fmt.Sprintf("./fixtures:%s", path))
```
Comment (carpeliam) [inline git_test.go]: @Chadius why this change? I was looking around the internet for what people do about the `os.Setenv` return value and I see examples of both assigning it to `_` and ignoring it.

File: git_tracker_name_translator.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+import (
+	"regexp"
+	"strconv"
+)
+
+// GetPivotalTrackerTaskID returns a string containing the task ID of the current branch name of the repository.
+func GetPivotalTrackerTaskID(branchName string) int {
+	re := regexp.MustCompile(`[1-9]+$`)
+	taskIdString := re.FindString(branchName)
```
Comment (carpeliam) [inline git_tracker_name_translator.go]: When I run `golint`, I get:

> git_tracker_name_translator.go:11:2: var taskIdString should be taskIDString
git_tracker_name_translator.go:12:2: var taskId should be taskID

File: git.go (hunk 1 above)
Comment (Chadius) [reply git.go]: I'm just gonna switch to VS Code- JetBrains's linter just disagrees.

File: git_tracker_name_translator.go [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package storybranch
+
+import (
+	"regexp"
+	"strconv"
+)
+
+// GetPivotalTrackerTaskID returns a string containing the task ID of the current branch name of the repository.
+func GetPivotalTrackerTaskID(branchName string) int {
+	re := regexp.MustCompile(`[1-9]+$`)
```
Comment (carpeliam) [inline git_tracker_name_translator.go]: It just occurred to me that a zero probably wouldn't be accepted here. Perhaps `\d+$`?

File: git.go (hunk 1 above)
Comment (carpeliam) [reply git.go]: Strange! `golint` would be the definitive linter here with regard to linters for go, so whatever `golint` says, that's the right thing. VS Code uses `golint` :) I use VS Code personally for go, though it tends to have some problems with variable renaming and other things, I imagine I just haven't spent enough time with the setup.

**Comment by Chadius** [comment]
Hey @carpeliam , I made a MockPivotalTrackerStoryService that follows the GetByID call that pivotal.StoryService uses to get the Story.

In the test, I can't use my mock directly into the Tracker, even though it wants an object that uses the interface. I have to do this workaround first. Any idea why?

```
		// TODO Why can't I pass in storyService directly?
		var i PivotalTrackerStoryService
		i = storyService
		gub := Tracker{&i}
```
